File: freecodecamp/bonfire/15-seek-and-destroy.rb
def destroyer(array, *args)
  array.delete_if { |item| args.include? item }
end

p destroyer([1,2,3,1,2,3], 2, 3) #=> [1, 1]
p destroyer([1, 2, 3, 5, 1, 2, 3], 2, 3) #=> [1, 5, 1]
p destroyer([3, 5, 1, 2, 2], 2, 3, 5) #=> [1]
p destroyer([2, 3, 2, 3], 2, 3) #=> []
p destroyer(["tree", "hamburger", 53], "tree", 53) #=> ["hamburger"]

File: freecodecamp/bonfire/15-seek-and-destroy.js
function destroyer(arr) {
  var destArgs = arguments;
  return arr.filter(function(el) {
    var match = false;
    for (var i = 1; i < destArgs.length; i++) {
      if (el == destArgs[i]) {
        match = true;
      }
    }
    return !match;
  });
}

console.log(destroyer([1, 2, 3, 1, 2, 3], 2, 3)); // [1, 1].
console.log(destroyer([1, 2, 3, 5, 1, 2, 3], 2, 3)); // [1, 5, 1].
console.log(destroyer([3, 5, 1, 2, 2], 2, 3, 5)); // [1].
console.log(destroyer([2, 3, 2, 3], 2, 3)); // [].
console.log(destroyer(["tree", "hamburger", 53], "tree", 53)); // ["hamburger"].
